components/camel-snmp/src/main/java/org/apache/camel/component/snmp/SnmpProducer.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.snmp;
+
+import java.util.concurrent.TimeoutException;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.impl.DefaultProducer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.snmp4j.CommunityTarget;
+import org.snmp4j.PDU;
+import org.snmp4j.Snmp;
+import org.snmp4j.TransportMapping;
+import org.snmp4j.event.ResponseEvent;
+import org.snmp4j.mp.MPv3;
+import org.snmp4j.security.SecurityModels;
+import org.snmp4j.security.SecurityProtocols;
+import org.snmp4j.security.USM;
+import org.snmp4j.smi.Address;
+import org.snmp4j.smi.GenericAddress;
+import org.snmp4j.smi.OID;
+import org.snmp4j.smi.OctetString;
+import org.snmp4j.smi.VariableBinding;
+import org.snmp4j.transport.DefaultTcpTransportMapping;
+import org.snmp4j.transport.DefaultUdpTransportMapping;
+
+/**
+ * A snmp producer
+ * 
+ *
+ */
+public class SnmpProducer extends DefaultProducer {
+   
+    private static final Logger LOG = LoggerFactory.getLogger(SnmpProducer.class);
+    
+    private SnmpEndpoint endpoint;
+    
+    private Address targetAddress;
+    private USM usm;
+
+    public SnmpProducer(SnmpEndpoint endpoint) {
+        super(endpoint);
+        this.endpoint = endpoint;
+    }
+    
+    @Override
+    public void start() throws Exception {
+        super.start();
+
+        this.targetAddress = GenericAddress.parse(endpoint.getAddress());
+        LOG.debug("targetAddress: {}", targetAddress);
+
+        this.usm = new USM(SecurityProtocols.getInstance(), new OctetString(MPv3.createLocalEngineID()), 0);
+        SecurityModels.getInstance().addSecurityModel(this.usm);
+    }
+    
+    @Override
+    public void stop() throws Exception {
+        super.stop();
+
```

Comment:
I wrote this producer 6 months ago... i remember that without this code i had an failure using snmpVersion2! you can found this code in more online examples!